libs/service/service.go

```diff
@@ -104,10 +104,6 @@ type BaseService struct {
 
 // NewBaseService creates a new BaseService.
 func NewBaseService(logger log.Logger, name string, impl Implementation) *BaseService {
-	if logger == nil {
```

Comment:
Since we're disallowing unset loggers, this could probably be a productive place for a panic. YMMV.

Reply:
I think it'll panic pretty quick (and te tests would catch this..) also there's no place in the code where we pass a nil logger, so I feel like it's ok to just let the panic happen for a nil pointer exception when we get there.